hyperparameter_tuning/service.py

```diff
@@ -0,0 +1,124 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from http.server import BaseHTTPRequestHandler, HTTPServer
+import re
+import cgi
+import json
+import requests
+import threading
+import time
+from urllib.parse import urlparse, parse_qs
+
+from tunables import get_all_tunables
+
+from bayes_optuna import optuna_hpo
+
+autotune_object_ids = []
+
+
+class HTTPRequestHandler(BaseHTTPRequestHandler):
+    def _set_response(self, status_code, return_value):
+        # TODO: add status_message
+        self.send_response(status_code)
+        self.send_header('Content-type', 'text/html')
+        self.end_headers()
+        self.wfile.write(return_value.encode('utf-8'))
+
+    def do_POST(self):
+        # TODO: perform better regex search
+        if re.search('/experiment_trials', self.path):
+            content_type, params = cgi.parse_header(self.headers.get('content-type'))
+            if content_type == 'application/json':
+                length = int(self.headers.get('content-length'))
+                str_object = self.rfile.read(length).decode('utf8')
+                json_object = json.loads(str_object)
+                # TODO: validate structure of json_object for each operation
+                if json_object["operation"] == "EXP_TRIAL_GENERATE_NEW":
+                    if json_object["id"] not in autotune_object_ids:
+                        autotune_object_ids.append(json_object["id"])
+                        get_search_create_study(json_object["id"], json_object["operation"], json_object["url"])
+                        trial_number = optuna_hpo.TrialDetails.trial_number
+                        self._set_response(200, str(trial_number))
+                    else:
+                        self._set_response(400, "-1")
+                elif json_object["operation"] == "EXP_TRIAL_GENERATE_SUBSEQUENT":
+                    if json_object["id"] in autotune_object_ids:
+                        get_search_create_study(json_object["id"], json_object["operation"], json_object["url"])
+                        trial_number = optuna_hpo.TrialDetails.trial_number
+                        self._set_response(200, str(trial_number))
+                    else:
+                        self._set_response(400, "-1")
+                elif json_object["operation"] == "EXP_TRIAL_RESULT":
+                    if json_object["id"] in autotune_object_ids and json_object["trial_number"] == optuna_hpo.TrialDetails.trial_number:
```

Comment:
We cannot be referencing optuna_hpo directly here. We could be using any algo. I would suggest that we have a separate function to get the number to which you pass the id. It should check the algo that is being used for that id and accordingly get the trial_number,